lib/wicked/IPv6.pm

```diff
@@ -0,0 +1,127 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper module for IPv6 tests
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+package wicked::IPv6;
+
+use base 'wickedbase';
+use testapi;
+use strict;
+use warnings;
+
+sub setup_interface {
+    my ($self, $iface, $ifcfg_type) = @_;
+    record_info("STEP", "Setup $ifcfg_type interface $iface");
+    $self->get_from_data('wicked/' . $ifcfg_type . '_address/ifcfg-eth0', '/etc/sysconfig/network/ifcfg-' . $iface);
+    $self->wicked_command('ifup', $iface);
+}
+
+sub wait_for_router_announcement {
+    my ($self, $iface) = @_;
+    record_info("STEP", "I let $iface wait for a router announcement");
+    assert_script_run('rdisc6 ' . $iface);
+}
+
+sub retrigger_dhcp_requests {
+    my ($self, $iface) = @_;
+    record_info("STEP", 'I retrigger DHCP requests');
+    $self->wicked_command('ifdown', $iface);
+    $self->wicked_command('ifup',   $iface);
+}
+
+
+sub obtain_address_by_dhcpv4 {
+    my ($self, $iface, $expected) = @_;
+    record_info("STEP", 'I ' . ($expected == 1 ? 'should' : 'should not') . ' obtain an address by DHCPv4');
+    # ip address show dev eth0
```

Comment:
I will remove this comments later on, but just for you to see what these actions mean.